File: 100dynamic/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        if (n == 1):
            return nums[0]
        tmp = nums[0]
        res = tmp
        for i in range(1,n):
            tmp = max(tmp + nums[i], nums[i])
            if tmp > res:
                res = tmp
        return res

File: 100dynamic/53.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut tmp: i32 = nums[0];
        let mut res: i32 = tmp;
        let n:usize = nums.len();
        for i in 1..n {
            tmp = std::cmp::max(tmp + nums[i as usize], nums[i as usize]);
            if tmp > res {
                res = tmp;
            }
        }
        res
    }
}